### PR TITLE
[ios] Enable training package in packaging pipeline

### PR DESCRIPTION
**Pull request by edgchen1**
### Description

Build iOS training package in packaging pipeline.
Refactor iOS packaging pipeline to build different package variants in parallel.

### Motivation and Context

Build iOS training package.

### CODE REVIEW

File: tools/ci_build/github/apple/default_training_ios_framework_build_settings.json [hunk 1]
```diff
@@ -14,6 +14,8 @@
         "--use_xcode",
         "--enable_training_apis",
         "--build_apple_framework",
+        "--use_coreml",
+        "--use_xnnpack",
```
Comment (edgchen1) [inline tools/ci_build/github/apple/default_training_ios_framework_build_settings.json:18]: @baijumeswani @askhade adding these to the training package build as they might be useful for inference. let me know if that's ok.

**Comment by edgchen1** [comment]
FYI, this change removes the actual upload of the pod archive files from the packaging pipeline. I am planning to move that to the release pipeline.

File: tools/ci_build/github/azure-pipelines/mac-ios-packaging-pipeline.yml [hunk 2]
```diff
@@ -1,235 +1,117 @@
 parameters:
-- name: BuildType
+- name: buildType
   displayName: |-
     Type of build.
     "release": A release build to be published for an official ONNX Runtime release.
-    "prerelease": A pre-release build to be published for validation prior to release.
```
[the remark this answers is not in the document]
Comment (edgchen1) [reply tools/ci_build/github/azure-pipelines/mac-ios-packaging-pipeline.yml:6]: we don't. before the only difference between "normal" and "prerelease" was whether the pod archive upload and podspec update would happen in this pipeline. but now I plan to move that step to the release pipeline. we can publish a pre-release version from a "normal" build's artifacts.

File: tools/ci_build/github/azure-pipelines/mac-ios-packaging-pipeline.yml [hunk 3]
```diff
@@ -1,235 +1,117 @@
...
+
+        set_var "ORT_POD_VERSION" "${VERSION}"
+      displayName: "Set common variables"
+      name: SetCommonVariables
+
+- template: templates/stages/mac-ios-packaging-build-stage.yml
+  parameters:
+    packageVariant: Mobile
+
+- template: templates/stages/mac-ios-packaging-build-stage.yml
```
[the remark this answers is not in the document]
Comment (edgchen1) [reply tools/ci_build/github/azure-pipelines/mac-ios-packaging-pipeline.yml:67]: the artifacts from the different build stages have different names (suffix "_<variant>"). you can specify an artifact by name for the whole pipeline. I'm not aware of a way to get an artifact for a specific stage.

**Review by YUNQIUGUO — APPROVED** [review]

**Review by skottmckay — APPROVED** [review]

**Pull request merged**